src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -0,0 +1,253 @@
+    kBoundsExceeded         = 2,
+    kNetworkIDNotFound      = 3,
+    kDuplicateNetworkID     = 4,
+    kNetworkNotFound        = 5,
+    kRegulatoryError        = 6,
+    kAuthFailure            = 7,
+    kUnsupportedSecurity    = 8,
+    kOtherConnectionFailure = 9,
+    kIPV6Failed             = 10,
+    kIPBindFailed           = 11,
+    kLabel9                 = 12,
+    kLabel10                = 13,
+    kLabel11                = 14,
+    kLabel12                = 15,
+    kLabel13                = 16,
+    kLabel14                = 17,
+    kLabel15                = 18,
+    kUnknownError           = 19,
+};
+
+enum class NetworkType : uint8_t
+{
+    kUndefined = 0,
+    kWiFi      = 1,
+    kThread    = 2,
+    kEthernet  = 3,
+};
+
+enum class WiFiSecurityType : uint8_t
+{
+    kUnencrypted  = 0,
+    kWEPPersonal  = 1,
+    kWPAPersonal  = 2,
+    kWPA2Personal = 3,
+    kWPA3Personal = 4,
+};
+
+struct ThreadNetworkInfo
+{
+    uint8_t mDataset[kMaxThreadDatasetLen];
+    uint8_t mDatasetLen;
+};
+
+struct WiFiNetworkInfo
+{
+    uint8_t mSSID[kMaxWiFiSSIDLen + 1];
+    uint8_t mSSIDLen;
+    uint8_t mCredentials[kMaxWiFiCredentialsLen];
+    uint8_t mCredentialsLen;
+    WiFiSecurityType mSecurity;
+};
+
+struct NetworkInfo
+{
+    uint8_t mNetworkID[kMaxNetworkIDLen];
+    uint8_t mNetworkIDLen;
+    uint8_t mEnabled;
+    NetworkType mNetworkType;
+    union NetworkData
+    {
+        ThreadNetworkInfo mThread;
+        WiFiNetworkInfo mWiFi;
+    } mData;
+};
+
+namespace {
+NetworkInfo sNetworks[kMaxNetworks];
+}
+
+void OnAddThreadNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                       uint32_t timeoutMs)
+{
+    CHIP_ERROR err                = CHIP_ERROR_NO_MEMORY;
+    size_t operationalDataset_Len = strlen(reinterpret_cast<const char *>(operationalDataset));
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDataset_Len);
+            sNetworks[i].mData.mThread.mDatasetLen = operationalDataset_Len;
+
+            sNetworks[i].mNetworkType = NetworkType::kThread;
+            sNetworks[i].mEnabled     = false;
+
+            memcpy(sNetworks[i].mNetworkID, kThreadNetworkName, sizeof(kThreadNetworkName));
+            sNetworks[i].mNetworkIDLen = sizeof(kThreadNetworkName);
+
+            err = CHIP_NO_ERROR;
+            break;
+        }
+    }
+
+    SuccessOrExit(err);
+exit:
+    ChipLogFunctError(err);
+}
+
+void OnAddWiFiNetworkCommandCallback(app::Command *, EndpointId, const uint8_t * ssid, const uint8_t * credentials,
+                                     uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR err         = CHIP_ERROR_NO_MEMORY;
+    size_t ssid_Len        = strlen(reinterpret_cast<const char *>(ssid));
```

Comment:
nit: can declare these where needed, to reduce scoping (more specific locality is easier to read)

Reply:
Thanks.